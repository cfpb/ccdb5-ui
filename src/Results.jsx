import { connect } from 'react-redux'
import ListPanel from './ListPanel'
import MapPanel from './MapPanel'
import React from 'react'
import TrendsPanel from './TrendsPanel'

export class Results extends React.Component {
  render() {
    let currentPanel

    switch ( this.props.tab ) {
      case 'Trends':
        currentPanel = <TrendsPanel/>
        break;
      case 'List':
        currentPanel = <ListPanel/>
        break;
      case 'Map':
      default:
        currentPanel = <MapPanel/>
        break;
    }

    return (
      <div className="content_main">
        { currentPanel }
      </div>
    )
  }
}

const mapStateToProps = state => ( {
  tab: state.view.tab
} )

export default connect( mapStateToProps )( Results )
