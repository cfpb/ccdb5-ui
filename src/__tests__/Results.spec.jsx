import React from 'react';
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { IntlProvider } from 'react-intl';
import Results from '../Results';
import renderer from 'react-test-renderer';

const fixture = [
  {
    company: 'foo',
    company_public_response: 'Closed',
    company_response: 'Closed',
    complaint_id: '1',
    complaint_what_happened: 'Lorem Ipsum',
    consumer_consent_provided: 'Yes',
    consumer_disputed: 'No',
    date_received: '2013-02-03T12:00:00Z',
    date_sent_to_company: '2013-01-01T12:00:00Z',
    issue: 'Foo',
    product: 'Bar',
    state: 'DC',
    sub_issue: 'Baz',
    sub_product: 'Qaz',
    submitted_via: 'email',
    timely: 'yes',
    zip_code: '200XX'
  }
]

function setupSnapshot(items=[], initialStore={}, tab = 'List') {
  const results = Object.assign({
    doc_count: 100,
    error: '',
    hasDataIssue: false,
    isDataStale: false,
    items,
    total: items.length
  }, initialStore)

  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    map: [],
    results,
    query: {},
    view: {
      tab: tab
    }
  });

  return renderer.create(
    <Provider store={ store } >
      <IntlProvider locale="en">
        <Results />
      </IntlProvider>
    </Provider>
  )
}

describe('component:Results', () => {
  it('renders map without crashing', () => {
    const target = setupSnapshot( fixture, null, 'Map' );
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders list without crashing', () => {
    const target = setupSnapshot( fixture, null, 'List' );
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });

})
