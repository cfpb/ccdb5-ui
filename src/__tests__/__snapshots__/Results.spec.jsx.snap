// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:Results renders list without crashing 1`] = `
<div
  className="content_main"
>
  <section
    className="results-panel"
  >
    <summary
      className="action-bar"
    >
      <div>
        <h2>
          Showing 
          <span>
            1
          </span>
           matches out of 
          <span>
            100
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <div
          className="cf-select flex-fixed"
        >
          <select
            id="choose-size"
            onChange={[Function]}
          >
            <option
              value={10}
            >
              Show 
              10
               results
            </option>
            <option
              value={25}
            >
              Show 
              25
               results
            </option>
            <option
              value={50}
            >
              Show 
              50
               results
            </option>
            <option
              value={100}
            >
              Show 
              100
               results
            </option>
          </select>
        </div>
        <div
          className="cf-select flex-fixed"
        >
          <select
            id="choose-sort"
            onChange={[Function]}
          >
            <option
              value="created_date_desc"
            >
              Sort by newest to oldest
            </option>
            <option
              value="created_date_asc"
            >
              Sort by oldest to newest
            </option>
            <option
              value="relevance_desc"
            >
              Sort by relevance
            </option>
            <option
              value="relevance_asc"
            >
              Sort by relevance (asc)
            </option>
          </select>
        </div>
        <h4
          className="flex-all"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export results
          </button>
        </h4>
      </div>
    </summary>
    <div />
    <ul
      className="cards-panel"
    >
      <li
        className="card-container"
      >
        <div
          className="card"
        >
          <div
            className="card-left layout-column"
          >
            <h3
              className="to-detail"
            >
              <a
                href="detail/1"
              >
                1
              </a>
            </h3>
            <h4>
              Company name
            </h4>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "foo",
                }
              }
            />
            <br />
            <h4>
              Company response to consumer
            </h4>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Closed",
                }
              }
            />
            <br />
            <h4>
              Timely response?
            </h4>
            <span
              className="body-copy"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "yes",
                }
              }
            />
          </div>
          <div
            className="card-right layout-column"
          >
            <div
              className="layout-row"
            >
              <div
                className="layout-row"
              >
                <h4>
                  Date received:
                </h4>
                <span
                  className="body-copy"
                >
                  <span>
                    2/3/2013
                  </span>
                </span>
              </div>
              <div
                className="spacer"
              />
              <div
                className="layout-row"
              >
                <h4>
                  Consumer's state:
                </h4>
                <span
                  className="body-copy"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "DC",
                    }
                  }
                />
              </div>
            </div>
            <br />
            <h4>
              Product
            </h4>
            <h3
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Bar",
                }
              }
            />
            <div
              className="layout-row"
            >
              <span
                className="body-copy subitem"
              >
                Sub-product:
              </span>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Qaz",
                  }
                }
              />
            </div>
            <br />
            <h4>
              Issue
            </h4>
            <h3
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Foo",
                }
              }
            />
            <div
              className="layout-row"
            >
              <span
                className="body-copy subitem"
              >
                Sub-issue:
              </span>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Baz",
                  }
                }
              />
            </div>
            <div>
              <br />
              <h4>
                Consumer Complaint Narrative
              </h4>
              <span
                className="body-copy"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Lorem Ipsum",
                  }
                }
              />
            </div>
          </div>
        </div>
      </li>
    </ul>
    <nav
      aria-label="Pagination"
      className="m-pagination"
      role="navigation"
    >
      <button
        className="a-btn m-pagination_btn-prev"
        disabled={false}
        onClick={[Function]}
      >
        <span
          className="a-btn_icon a-btn_icon__on-left"
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 559.6 1200"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M494.5 1090.7c-17.3 0-33.8-6.8-46-19L19 642.1c-25.4-25.4-25.4-66.5 0-91.9l429.5-429.5c25.6-25.1 66.8-24.8 91.9.8 24.8 25.3 24.8 65.8 0 91.1L156.9 596.2l383.6 383.6c25.4 25.4 25.4 66.5.1 91.9-12.3 12.2-28.8 19-46.1 19z"
            />
          </svg>
        </span>
        Previous
      </button>
      <button
        className="a-btn m-pagination_btn-next"
        disabled={false}
        onClick={[Function]}
      >
        Next
        <span
          className="a-btn_icon a-btn_icon__on-right"
        >
          <svg
            className="cf-icon-svg"
            viewBox="0 0 559.6 1200"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M65.1 1090.2c-35.9 0-65-29-65.1-64.9 0-17.3 6.8-33.9 19.1-46.1l383.6-383.5L19.1 212.2c-25.1-25.6-24.8-66.8.9-92 25.3-24.8 65.8-24.8 91.1 0l429.5 429.5c25.4 25.4 25.4 66.5 0 91.9L111 1071.2c-12.1 12.2-28.7 19.1-45.9 19z"
            />
          </svg>
        </span>
      </button>
      <form
        action=""
        className="m-pagination_form"
        onSubmit={[Function]}
      >
        <label
          className="m-pagination_label"
          htmlFor="m-pagination_current-page"
        >
          Page
          <span
            className="u-visually-hidden"
          >
            number 
             out
          </span>
          <input
            className="m-pagination_current-page"
            id="m-pagination_current-page"
            inputMode="numeric"
            max={1}
            min="1"
            name="page"
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
          />
          of 
          <span>
            1
          </span>
        </label>
        <button
          className="a-btn a-btn__link m-pagination_btn-submit"
          id="m-pagination_btn-submit"
          type="submit"
        >
          Go
        </button>
      </form>
    </nav>
  </section>
</div>
`;

exports[`component:Results renders map without crashing 1`] = `
<div
  className="content_main"
>
  <section
    className="map-panel"
  >
    <summary
      className="action-bar"
    >
      <div>
        <h2>
          Showing 
          <span>
            1
          </span>
           matches out of 
          <span>
            100
          </span>
           total complaints
        </h2>
      </div>
      <div
        className="layout-row"
      >
        <h4
          className="flex-all"
        >
          <button
            className="a-btn a-btn__link"
            data-gtm_ignore="true"
            onClick={[Function]}
          >
            Export results
          </button>
        </h4>
      </div>
    </summary>
    <div />
    <section
      className="date-intervals"
    >
      <button
        className="date-selector "
        onClick={[Function]}
      >
        3m
      </button>
      <button
        className="date-selector "
        onClick={[Function]}
      >
        6m
      </button>
      <button
        className="date-selector "
        onClick={[Function]}
      >
        1y
      </button>
      <button
        className="date-selector "
        onClick={[Function]}
      >
        3y
      </button>
      <button
        className="date-selector "
        onClick={[Function]}
      >
        All
      </button>
    </section>
    <div>
      <div
        className="cfpb-chart"
        data-chart-type="tile_map"
        id="tile-chart-map"
      />
    </div>
  </section>
</div>
`;
